tests/Services/SearchServiceTest.php

```diff
@@ -332,154 +337,893 @@ public function categorySearchProvider(): array
+                                    ]
+                                ]
+                            ]
+                        ],
+                        [
+                            'id' => 1012,
+                            'data' => [
+                                'texts' => [
+                                    [
+                                        'urlPath' => 'test-product'
+                                    ]
+                                ],
+                                'item' => [
+                                    'id' => 11
+                                ],
+                                'variation' => [
+                                    'id' => 1012,
+                                    'isMain' => false,
+                                    'model' => 'model',
+                                    'number' => 'number',
+                                    'order' => 'this is the text that was searched for'
+                                ],
+                                'salesPrices' => [
+                                    [
+                                        'price' => 0.00
+                                    ]
+                                ]
+                            ],
+                        ],
+                        [
+                            'id' => 1013,
+                            'data' => [
+                                'texts' => [
+                                    [
+                                        'urlPath' => 'test-product'
+                                    ]
+                                ],
+                                'item' => [
+                                    'id' => 11
+                                ],
+                                'variation' => [
+                                    'id' => 1013,
+                                    'isMain' => false,
+                                    'model' => 'model',
+                                    'number' => 'number',
+                                    'order' => 'this is the text that was searched for'
+                                ],
+                                'salesPrices' => [
+                                    [
+                                        'price' => 0.00
+                                    ]
+                                ]
+                            ],
+                        ]
+                    ]
+                ],
+                'shopUrl' => 'https://www.test.com',
+                'dataQueryInfoMessage' => [
+                    'queryStringType' => 'notImprovedOrCorrected'
+                ],
+                'redirectUrl' => '/test-product_11',
+                'attributes' => [
+                    'page' => 1
+                ],
+                'de',
+                'de',
+                false
+            ],
+            'One product with three variations, main variation without price, no query matches' => [
```

Comment:
Since almost all tests create three similar variation, maybe you can combine the method from above with an additional helper that creates a set amount of variations, based on data for each variation.